src/tests/JIT/opt/InstructionCombining/AcrossAndCselToAcross.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// unit test for the full range comparison optimization
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+public class AcrossAndCselToAcross
+{
+
+    [Theory]
+    [InlineData(1, 42)]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static Vector<long> addAcross_sbyte(sbyte op1, sbyte op2)
+    {
+        //ARM64-FULL-LINE: cmpne {{p[0-9]+.b}}, {{p[0-9]+/z}}, {{z[0-9]+.b}}, #0
+        //ARM64-FULL-LINE-NEXT: saddv {{d[0-9]+}}, {{p[0-9]+}}, {{z[0-9]+.b}}
+        var vec = Vector.Create(op2);
+        var mask = Vector.Create(op1);
+        return Sve.AddAcross(Sve.ConditionalSelect(mask, vec, Vector<sbyte>.Zero));
+    }
+
+    [Theory]
+    [InlineData(1, 42, 43)]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static Vector<double> addSequentialAcross_double(double op1, double op2, double op3)
+    {
+        //ARM64-FULL-LINE: cmpne {{p[0-9]+.d}}, {{p[0-9]+/z}}, {{z[0-9]+.d}}, #0
+        //ARM64-FULL-LINE-NEXT: fadda {{v[0-9]+.d}}, {{p[0-9]+}}, {{v[0-9]+.d}}, {{z[0-9]+.d}}
+        var vec1 = Vector.Create(op2);
+        var vec2 = Vector.Create(op3);
+        var mask = Vector.Create(op1);
+        return Sve.AddSequentialAcross(vec1, Sve.ConditionalSelect(mask, vec2, Vector<double>.Zero));
```

Comment:
can we also add a scenario for `AllBitsSet`, `One` and `vec2` or something for completion?

Reply:
Updated one of the tests to take the value with all 1s as input. 👍